utils/mgmt_system/google.py

```diff
@@ -0,0 +1,290 @@
+        'starting': ('STAGING'),
+        'stopping': ('STOPPING'),
+    }
+
+    def __init__(self, project=None, zone=None, default_instance_name=None, scope=None,
+            oauth2_storage=None, client_secrets=None):
+
+        self._project = project
+        self._zone = zone
+
+        # Default name of the instance name from config file
+        self._instance_name = default_instance_name
+
+        # Perform OAuth 2.0 authorization.
+        # based on OAuth 2.0 client IDs credentials from client_secretes file
+        if client_secrets and scope and oauth2_storage:
+            flow = flow_from_clientsecrets(client_secrets, scope=scope)
+            storage = Storage(oauth2_storage)
+            self._credentials = storage.get()
+
+        if self._credentials is None or self._credentials.invalid:
+            self._credentials = run_flow(flow, storage, argparser.parse_args([]))
+
+        if self._credentials is None or self._credentials.invalid:
+            raise Exception("Incorrect credentials for Google Cloud System")
+
+        self._compute = build('compute', 'v1', credentials=self._credentials)
+
+    def _get_all_instances(self):
+        return self._compute.instances().list(project=self._project, zone=self._zone).execute()
+
+    def list_vm(self):
+        result = self._get_all_instances()
+        return result['items']
+
+    def _find_instance_by_name(self, instance_name):
+        try:
+            instance = self._compute.instances().get(
+                project=self._project, zone=self._zone, instance=instance_name).execute()
+            return instance
+        except Exception:
+            raise VMInstanceNotFound(instance_name)
+
+    def _nested_wait_vm_running(self, operation_name):
+        result = self._compute.zoneOperations().get(
+            project=self._project,
+            zone=self._zone,
+            operation=operation_name).execute()
+
+        if result['status'] == 'DONE':
+            logger.info("The operation %s -> DONE" % operation_name)
+            return True
+            if 'error' in result:
+                logger.error("Error during {} operation.".format(operation_name))
+                raise Exception(result['error'])
+
+        return False
+
+    def create_vm(self, instance_name=None, source_disk_image=None, machine_type=None,
+            startup_script=None, timeout=180):
+        if self.does_vm_exist(instance_name):
+            logger.info(" The %s instance is already exists, skipping" % instance_name)
+            return True
+
+        logger.info("Creating %s instance" % instance_name)
+        if not instance_name:
+            instance_name = self._instance_name
+
+        if not source_disk_image:
+            source_disk_image = "projects/debian-cloud/global/images/debian-7-wheezy-v20150320"
+
+        machine_type = machine_type or ("zones/%s/machineTypes/n1-standard-1" % self._zone)
+
+        try:
+            script = open(startup_script, 'r').read()
```

Comment:
We should probably look at using the path module for this, https://github.com/RedHatQE/cfme_tests/blob/master/cfme/infrastructure/pxe.py#L587
I think actually, the ```create_vm``` function shouldn't be concerned with loading the data from the file and instead should take the actual script data instead of the script name